### ISSUE TITLE
Consider moving from `java-json-tools/json-schema-validator` to `networknt/json-schema-validator`

### ISSUE THREAD

**Issue by lbroudoux** [microcks/microcks#1396]
### Reason/Context

As of today, we used `java-json-tools/json-schema-validator` for all the things related to JSON schema validation in OpenAPI, AsyncAPI and GraphQL schema tests. However, this library seems to be no longer maintained with the latest release being more that 4 years old (see https://github.com/java-json-tools/json-schema-validator).

### Description

New Java Json Schema validation implementations have been started since then and are in active development. The JSON Schema website list a few of them [here](https://json-schema.org/tools?query=&sortBy=name&sortOrder=ascending&groupBy=toolingTypes&licenses=&languages=Java&drafts=&toolingTypes=validator&environments=).

According to Creekservice Performance Benchmark (see [here](https://www.creekservice.org/json-schema-validation-comparison/performance)) and Bowtie service ran by the JSON Schema community (see [here](https://bowtie.report/#/?language=java)), the [`networknt/json-schema-validator`](https://github.com/networknt/json-schema-validator) seems to be a very good candidate with recent releases, very active community and compatible license.

We should give it a try as a replacement for the current solution and check if there are performance benefits.

**Comment by Harsh4902** [comment]
@lbroudoux I want to work on this issue.

**Comment by lbroudoux** [comment]
Awesome! Do you need some guidance?

**Comment by Harsh4902** [comment]
@lbroudoux I did some research on the classes of `java-json-tools/json-schema-validator ` which we are using in our code, and tried to find classes with similar functionality from `networknt/json-schema-validator`. I am not able to find similar classes for `ProcessingMessage` and `ProcessingReport`. Can you guide me on this?

**Comment by lbroudoux** [comment]
Hi @Harsh4902 
Any news on a GitHub repo or Gist you initialized somewhere for this issue?

**Comment by Harsh4902** [comment]
Hi @lbroudoux yes, [here](https://github.com/Harsh4902/microcks/tree/issue%231396) is the branch for this.

**Comment by lbroudoux** [comment]
Hey @Harsh4902 
I had a quick look at it and succeeded in having minimal examples working. It should obviously be re-worked to support more complex use cases (I think about integrating external references that are served internally by Microcks HTTP layer).

What I've done so far:

1. Change `extractJsonSchemaNode()` to

```java
private static JsonSchema extractJsonSchemaNode(JsonNode jsonNode, String namespace) {
   JsonSchemaFactory jsonSchemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012);
   return jsonSchemaFactory.getSchema(jsonNode);
}
```

2. Change `validationJson()` to

```java
public static List<String> validateJson(JsonNode schemaNode, JsonNode jsonNode, String namespace) {
   List<String> errors = new ArrayList<>();

   final JsonSchema jsonSchemaNode = extractJsonSchemaNode(schemaNode, namespace);

   Set<ValidationMessage> assertions = jsonSchemaNode.validate(jsonNode, executionContext -> {
      // By default since Draft 2019-09 the format keyword only generates annotations and not assertions
      executionContext.getExecutionConfig().setFormatAssertionsEnabled(true);
      executionContext.getExecutionConfig().setLocale(Locale.US);
   });
   for (ValidationMessage message : assertions) {
      errors.add(message.getMessage());
   }

   return errors;
}
```

3. Change `getJsonNode()` to

```java
public static JsonNode getJsonNode(String jsonText) throws IOException {
   return mapper.readTree(jsonText);
}
```

with `mapper` being:

```java
private static final ObjectMapper mapper = new ObjectMapper()
         .enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)
         .enable(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN).enable(SerializationFeature.INDENT_OUTPUT);
```

Validation messages in unit tests should be adapted though but the good news is that they're much more detailed than the one we got with the previous library.

Let me know if it helps!

**Comment by Harsh4902** [comment]
@lbroudoux Yes it helped a lot. I also changed validation messages in unit test for `JsonSchemaValidator`, and now trying to identify potential changes in other classes which are using `JsonSchemaValidator`. You can see changes [here](https://github.com/Harsh4902/microcks/commit/3a7ae70b332ac9137261946ed3cf664cc2b7cae5).

**Comment by lbroudoux** [comment]
Ok cool! Take of code style. You have to run "mvn spotless:apply" to ensure consistent formatting.

**Comment by Harsh4902** [comment]
@lbroudoux I have raised this draft PR to keep track of progress on this issue. I will squash all the commits once migration will complete.

**Comment by lbroudoux** [comment]
Now done!
